packages/theme-check-common/src/index.ts

```diff
@@ -74,6 +75,7 @@ export async function check(
     getDefaultTranslations: makeGetDefaultTranslations(fs, theme, rootUri),
     getDefaultSchemaLocale: makeGetDefaultSchemaLocale(fs, rootUri),
     getDefaultSchemaTranslations: makeGetDefaultSchemaTranslations(fs, theme, rootUri),
+    getMetafieldDefinitions: makeGetMetafieldDefinitions(fs),
```

Comment:
Since `makeGetMetafieldDefinitions` reads the file on the filesystem, wouldn't this actually run every time we run check?

I feel like we need to memoize it too (like we did in startServer)

Reply:
hmm yeah. I think you'll need to do two things here:
- to support caching in language server, we inject the memoized version of the function as an optional dep
- to support CLI use cases, we do makeGetMetafieldDefinitions(fs) as a fallback

this means:
```ts
interface Dependencies {
  getMetafieldDefinitions?: () => Promise<MetafieldDefinition[]>
}

// ... 

const dependencies: AugmentedDependencies = {
   ...injectedDependencies,
   // ...
   getmetafieldDefinitions = injectedDependencies.getMetafieldDefinitions ?? makeGetMetafieldDefinitions(fs),
}
```